packages/presets/src/ai/config/edgeless-copilot.ts

```diff
@@ -1,317 +1,172 @@
-  name: 'Create an image',
-  icon: AIPenIcon,
-  showWhen: (_, editorMode, host: EditorHost) => {
-    return (
-      editorMode === 'edgeless' &&
-      showWhen(host, service => {
-        const selected = getSelectedElements(service);
-
-        return (
-          selected.length === 0 ||
-          selected[0] instanceof ImageBlockModel ||
-          selected[0] instanceof NoteBlockModel ||
-          selected[0] instanceof TextElementModel
-        );
-      })
-    );
-  },
-};
-
-const createMindmap: AIItemConfig = {
-  name: 'Create a mindmap',
-  icon: AIPenIcon,
-  showWhen: (_, editorMode, host: EditorHost) => {
-    return (
-      editorMode === 'edgeless' &&
-      showWhen(host, service => {
-        const selected = getSelectedElements(service);
-
-        return (
-          selected.length === 0 ||
-          selected[0] instanceof ImageBlockModel ||
-          selected[0] instanceof TextElementModel
-        );
-      })
-    );
-  },
-  handler: host => {
-    const rootBlockId = host.doc.root?.id;
-    assertExists(rootBlockId);
-
-    const aiPanel = host.view.getWidget(AFFINE_AI_PANEL_WIDGET, rootBlockId);
-    const copilotPanel = host.view.getWidget(
-      AFFINE_EDGELESS_COPILOT_WIDGET,
-      rootBlockId
-    );
-    const selectedElement = getSelectedElements(getService(host))[0];
-
-    if (!(aiPanel instanceof AffineAIPanelWidget) || !selectedElement) return;
-
-    assertType<EdgelessCopilotWidget>(copilotPanel);
-    copilotPanel.hide();
-
-    const selectionRect = copilotPanel.selectionModelRect;
-
-    aiPanel.config = {
-      answerRenderer: createMindmapRenderer(host, aiPanel),
-      generateAnswer: () => {},
-
-      finishStateConfig: {
-        responses: [
-          {
-            name: 'Insert',
-            icon: InsertBelowIcon,
-            handler: () => {
-              const [surface] = host.doc.getBlockByFlavour('affine:surface');
-
-              assertType<SurfaceBlockModel>(surface);
-
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              const data = aiPanel.ctx as any;
-              aiPanel.hide();
-
-              const mindmapId = surface.addElement({
-                type: 'mindmap',
-                children: data.node,
-                style: data.style,
-              });
-              const mindmap = surface.getElementById(
-                mindmapId
-              ) as MindmapElementModel;
-
-              host.doc.transact(() => {
-                const rootElement = mindmap.tree.element;
-
-                if (selectionRect) {
-                  rootElement.xywh = `[${selectionRect.x},${selectionRect.y},${rootElement.w},${rootElement.h}]`;
-                }
-              });
-            },
-          },
-        ],
-        actions: [],
-      },
-      errorStateConfig: {
-        upgrade: () => {},
-        responses: [],
-      },
-    };
+import type { AIItemGroupConfig } from '@blocksuite/blocks';
```

Comment:
I think @Flrande mentioned that the edgless config should be located at `ai/entries/edgeless`

Reply:
We can move item components from blocks to preset

Reply:
Entries subfolders correspond to setup functions for each ui entry (such as listening for events to trigger panel when a space is entered)